### PR TITLE
Add trello and supabase tools

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: clis/supabase/README.md
Comment (g-linville) [inline clis/supabase/README.md:1]: This README can probably be deleted as it doesn't really provide much information.

File: clis/supabase/tool.gpt
Comment (g-linville) [inline clis/supabase/tool.gpt:1]: I recommend changing this to match how Taylor set up the tools for other CLIs in this repo.

File: apis/supabase/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  - File: `read-only/tool.gpt`
+  - Description: Helps with read operations in Supabase API.
+  - Tools: `v1-get*`, `v1-list*` from `openapi.yaml`.
+
+- **Read-Write Operations:**
+
+  - File: `read-write/tool.gpt`
+  - Description: Helps with read and write operations in Supabase API.
+  - Tools: All the tools from `openapi.yaml`.
```
Comment (g-linville) [inline apis/supabase/README.md:27]:
```suggestion
# Supabase API Tools

This folder contains tools for GPTScript to interact with the Supabase API.

## Prerequisites

Before you begin, make sure you have a Supabase API key. The instructions for creating one are [here](https://supabase.com/docs/reference/api/introduction).

## Available Tools

- github.com/gptscript-ai/tools/apis/supabase/read-only
  - This is a set of tools that can read information from the Supabase API, but will not change anything.
- github.com/gptscript-ai/tools/apis/supabase/read-write
  - This is a broader set of tools that can read, write, and delete things in the Supabase API.
```

File: apis/trello/tool.gpt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Before running sys-openapi tool, give an explaination on what it can do.
+When asked to get current org. use "me" as id.
+For other case, use the id from the context or ask user to provide ID.
+
+---
+Name: trello-api-key
+
+#!/usr/bin/env bash
+
+echo "{\"env\":{\"GPTSCRIPT_API_TRELLO_COM_APIKEY\":\"ef4fde9fd7f65a15c74322f7546e4dee\"}}"
```
Comment (g-linville) [inline apis/trello/tool.gpt:20]:
```suggestion
Name: Trello API
Share Credential: trello-api-key as api.trello.comAPIKey
Share Credential: github.com/gptscript-ai/credential as api.trello.comAPIToken with GPTSCRIPT_API_TRELLO_COM_APITOKEN as env and "Please go to [Trello](https://trello.com/1/authorize?expiration=never&scope=read,write,account&response_type=token&key=ef4fde9fd7f65a15c74322f7546e4dee) to create API token and provide it here" as message and key as field
Share Tools: *boards* from ./openapi.json
Share Tools: *cards* from ./openapi.json
Share Tools: *organizations* from ./openapi.json

---
Name: trello-api-key

#!/usr/bin/env bash

echo "{\"env\":{\"GPTSCRIPT_API_TRELLO_COM_APIKEY\":\"ef4fde9fd7f65a15c74322f7546e4dee\"}}"
```

File: apis/trello/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+To retrieve all boards in a specific organization, use the `get-organizations-id-boards` tool. You will need to provide the organization ID.
+
+#### Retrieve All Organizations a Member Belongs To
+
+To retrieve all organizations a member belongs to, use the `get-members-id-organizations` tool. You will need to provide the member ID. To get the current user's organizations, use "me" as the ID.
+
+## Contributing
+
+If you would like to contribute to this project, please fork the repository and submit a pull request.
```
Comment (g-linville) [inline apis/trello/README.md:61]:
```suggestion
# Trello API Tool

This is a set of GPTScript tools to interact with the Trello API.

## Prerequisite

You need generate a Trello API token [here](https://trello.com/1/authorize?expiration=never&scope=read,write,account&response_type=token&key=aabf1a5f6af0a2e5c4c3807b4d3ccbc8). You will be automatically prompted for this information when you are running the tool.

## Quick Start

Create a new file called `agent.gpt` with this as the contents:

```
Tools: github.com/gptscript-ai/tools/apis/trello
Chat: true

Please help me with my Trello workspace.
```

Then, run it with `gptscript agent.gpt`. You will be prompted for your API key, and after providing it, you will be able to chat with the large language model and have it perform actions for you in Trello.

## Tools

These are some of the tools available to the LLM when you use this tool set:

- **post-cards**: Create new cards on a Trello board.
- **delete-cards-id**: Delete a card by its ID.
- **get-boards-id-cards**: Retrieve all cards on a specific board.
- **get-organizations-id-boards**: Retrieve all boards in a specific organization.
- **get-members-id-organizations**: Retrieve all organizations a member belongs to.
```

**Review by g-linville — CHANGES_REQUESTED** [review]

File: apis/supabase/read-only/tool.gpt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name: Supabase API(Read)
+Credential: github.com/gptscript-ai/credential as api.supabase.combearer with GPTSCRIPT_API_SUPABASE_COM_BEARER as env and "Please go to instruction [here](https://supabase.com/docs/reference/api/introduction) and provide your supabase API key" as message and key as field
+Description: Allows user to run supabase API with read operations
+Tools: v1-get* from ./openapi.yaml
+Tools: v1-list* from ./openapi.yaml
+
+You are a helpful assistant helping me nagivating supabase openapi docs and give me instructions to how to use their api.
+
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply apis/supabase/read-only/tool.gpt:8]: does it need description though?

File: apis/supabase/read-write/tool.gpt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+Name: Supabase API(Read-Write)
+Credential: github.com/gptscript-ai/credential as api.supabase.combearer with GPTSCRIPT_API_SUPABASE_COM_BEARER as env and "Please go to instruction [here](https://supabase.com/docs/reference/api/introduction) and provide your supabase API key" as message and key as field
+Description: Allows user to run supabase API with read and write operations
+Tools: * from ./openapi.yaml
+
+You are a helpful assistant helping me nagivating supabase openapi docs and give me instructions to how to use their api.
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply apis/supabase/read-write/tool.gpt]: Deleted description and changed to `Share Credential` and `Share Tools`

File: clis/supabase/tool.gpt
Comment (StrongMonkey) [reply clis/supabase/tool.gpt:1]: Yeah changed 👍

**Review by g-linville — APPROVED** [review]

File: clis/supabase/tool.gpt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Name: Supabase CLI
+Type: context
+Share Tools: sys.exec
+Description: Helps user to interacte with supabase CLI
```
Comment (drpebcak) [inline clis/supabase/tool.gpt:4]:
```suggestion
Description: Helps user interact with the supabase CLI
```

File: apis/trello/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+Create a new file called `agent.gpt` with this as the contents:
+
+```
+Tools: github.com/gptscript-ai/tools/apis/trello
+Chat: true
+
+Please help me with my Trello workspace.
+```
+
+Then, run it with `gptscript agent.gpt`. You will be prompted for your API key, and after providing it, you will be able to chat with the large language model and have it perform actions for you in Trello.
```
Comment (drpebcak) [inline apis/trello/README.md:20]:
```suggestion
Then, run it with `gptscript agent.gpt`. You'll be prompted to enter your API key; once you provide it, you'll be able to chat with the large language model and have it perform actions for you in Trello.
```

File: apis/supabase/read-write/tool.gpt [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Name: Supabase API(Read-Write)
+Share Credential: github.com/gptscript-ai/credential as api.supabase.combearer with GPTSCRIPT_API_SUPABASE_COM_BEARER as env and "Please go to instruction [here](https://supabase.com/docs/reference/api/introduction) and provide your supabase API key" as message and key as field
```
Comment (drpebcak) [inline apis/supabase/read-write/tool.gpt:2]:
```suggestion
Share Credential: github.com/gptscript-ai/credential as api.supabase.combearer with GPTSCRIPT_API_SUPABASE_COM_BEARER as env and "Please read the instructions [here](https://supabase.com/docs/reference/api/introduction) and provide your supabase API key" as message and key as field
```

File: apis/supabase/read-only/tool.gpt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+Name: Supabase API(Read)
+Share Credential: github.com/gptscript-ai/credential as api.supabase.combearer with GPTSCRIPT_API_SUPABASE_COM_BEARER as env and "Please go to instruction [here](https://supabase.com/docs/reference/api/introduction) and provide your supabase API key" as message and key as field
```
Comment (drpebcak) [inline apis/supabase/read-only/tool.gpt:2]:
```suggestion
Share Credential: github.com/gptscript-ai/credential as api.supabase.combearer with GPTSCRIPT_API_SUPABASE_COM_BEARER as env and "Please read the instructions [here](https://supabase.com/docs/reference/api/introduction) and provide your supabase API key" as message and key as field
```

**Review by drpebcak — CHANGES_REQUESTED** [review]

File: apis/trello/tool.gpt [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+Share Tools: *boards* from ./openapi.json
+Share Tools: *cards* from ./openapi.json
+Share Tools: *organizations* from ./openapi.json
+
+---
+Name: trello-api-key
+
+#!/usr/bin/env bash
+
+echo "{\"env\":{\"GPTSCRIPT_API_TRELLO_COM_APIKEY\":\"ef4fde9fd7f65a15c74322f7546e4dee\"}}"
```
Comment (drpebcak) [inline apis/trello/tool.gpt:13]: Is this key meant to be in here?

File: apis/trello/tool.gpt [hunk 11]
```diff
@@ -0,0 +1,13 @@
+Name: Trello API
+Share Credential: trello-api-key as api.trello.comAPIKey
+Share Credential: github.com/gptscript-ai/credential as api.trello.comAPIToken with GPTSCRIPT_API_TRELLO_COM_APITOKEN as env and "Please go to [here](https://trello.com/1/authorize?expiration=never&scope=read,write,account&response_type=token&key=ef4fde9fd7f65a15c74322f7546e4dee) to authenticate and provide your trello API token" as message and key as field
```
Comment (drpebcak) [inline apis/trello/tool.gpt:3]:
```suggestion
Share Credential: github.com/gptscript-ai/credential as api.trello.comAPIToken with GPTSCRIPT_API_TRELLO_COM_APITOKEN as env and "Please go [here](https://trello.com/1/authorize?expiration=never&scope=read,write,account&response_type=token&key=ef4fde9fd7f65a15c74322f7546e4dee) to authenticate and provide your trello API token" as message and key as field
```

**Review by drpebcak — CHANGES_REQUESTED** [review]

File: apis/trello/tool.gpt (hunk 10 above)
Comment (StrongMonkey) [reply apis/trello/tool.gpt:13]: This is public API key and it is safe be to here. It identifies the auth app. 
>Tokens for users should always be securely stored as they grant access to the entire user's account! It is ok for your API key to be publicly available, but a token should never be publicly available. If a token becomes public, it should be revoked immediately by the user.

**Review by drpebcak — APPROVED** [review]

**Pull request merged**